elastic/kibana
# [ResponseOps] add license check for connector execution

resolves https://github.com/elastic/response-ops-team/issues/198

## Summary

Ensures that connector execution occurs based on current license level.

## To Verify

The basic idea is to create a > basic license connector in a trial license, and then downgrade the license to basic and see if that connector can still be run.  Any connector besides server log and index can be used - those are the only connectors licensed for basic.

Start ES as follows to run with a trial license:

    yarn es snapshot --license trial

Create a connector, and also create an alerting rule which uses the connector, and is arranged to always be active and alert on check intervals (not on status change).  Ensure the connector runs when the alert is active, and that you can run it in "test" mode from the connector page.

Then go here, to change your license to basic:

  http://localhost:5601/app/management/stack/license_management

You should now see the following sorts of messages logged by Kibana:

```
[ERROR][plugins.taskManager] Task actions:.email "a0df7b77-b6e5-461e-9699-178d6878f235" failed: Error: Action type .email is disabled because your basic license does not support it. Please upgrade your license.
[WARN ][plugins.alerting.index-threshold] Rule "c6780109-15b8-4816-ba96-266e4ddc482d" skipped scheduling action "gmail" because it is disabled
[ERROR][plugins.actions] Action 'gmail' failed: Action type .email is disabled because your basic license does not support it. Please upgrade your license.
```

When attenpting to "test" the connector, you should see the following result:

```
Test failed to run

The following error was found:
Forbidden
```

(we'll attempt to improve the message later as a follow-up)
